Biocompute Object export cleanup.

jmchilton opened the pull request:
See review of https://github.com/galaxyproject/galaxy/pull/9077/files

jmchilton committed:
Remove contributing user's heuristic in BioCompute Objects.

I think this needs to be tracked explicitly - see https://github.com/galaxyproject/galaxy/issues/10016.

jmchilton committed:
Some small cleanup on BCO export based Marius' review.

mvdbeek commented on lib/galaxy/webapps/galaxy/api/workflows.py:
```diff
@@ -1119,24 +1110,24 @@ def _generate_invocation_bco(self, trans, invocation_id, **kwd):
-                    'version': current_tool['tool_version'],
+                    'name': current_step['name'],
+                    'description': current_step['annotation'],
+                    'version': current_step['tool_version'],
```
I don't think steps have a tool_version, maybe skip this ? That should be what causes https://jenkins.galaxyproject.org/job/docker-api-py3/5470/testReport/junit/lib.galaxy_test.api.test_workflows/WorkflowsApiTestCase/test_run_subworkflow_simple/

HadleyKing replied:
From the little that I have seen the following line: https://github.com/galaxyproject/galaxy/blob/2f2923d4043dc305a664cb80c91d0a74e3dbec66/lib/galaxy/webapps/galaxy/api/workflows.py#L1077
caused this to work in the limited tests I was able to run. Even pulling in a `None` some times, but I guess that was because I was always using tools from the Tool Shed. For reproducibility the version is important. Do you think we could  find some workaround or alternate solution as opposed to skipping it?

mvdbeek replied:
My bad, steps may have a tool_version in the exported workflow dict (I'd personally would have worked with the workflow, but that's fine for now), but not if the step is not a tool step. The order of the steps was not correct though, so when you pick out the workflow_dict step using the index (i) of the workflow step you may or may not get the step you think you're getting, so if you pick a non tool step this errors out. I've fixed that in the last commit.

nsoranzo commented on lib/galaxy/webapps/galaxy/api/workflows.py:
```diff
@@ -1005,34 +1003,27 @@ def _generate_invocation_bco(self, trans, invocation_id, **kwd):
             versions.append(version)
+            if workflow == w:
+                break
         current_version = versions[-1]['version']
```
I think lines 1016-1027 can be greatly simplified to:
```python
        current_version = None
        for i, w in enumerate(reversed(stored_workflow.workflows)):
            if workflow == w:
                current_version = i
```

HadleyKing reviewed (commented):
RE: https://github.com/galaxyproject/galaxy/pull/10217/commits/6e790ae2048204093b00878fb31a1c9e36f7afc8 
Is there a replacement or are we waiting for a UI to complete these fields? Would that replacement be something that is an addition to the workflow creation/editing?

mvdbeek committed:
Fix a bunch of correctness issues in bco export

- urls are now correct
- step order is now correct (step_order is the authoritative order)
- collection ids should be correct
- return empty affiliation field (my affiliation is not gmail.com ...)
- return content_id as name, repository name alone is not helpful

I think at this state it should work for simple invocations,
but future iterations should include subworkflow and parameter
type steps.
